### DISCUSSION TITLE
Manually Installing DNS Monster with ClickHouse on AlmaLinux - Data Not Loading Issue

### DISCUSSION THREAD

**Discussion by pagliardinipert** [mosajjal/dnsmonster#101] [Q&A]
I've been working on setting up DNS Monster on AlmaLinux to integrate with ClickHouse for data storage. Despite configuring everything as per the setup instructions, I'm encountering an issue where no data is being loaded into ClickHouse.

Here are the steps I've followed:

Installation: I installed DNS Monster on AlmaLinux using Go and configured it to send data to ClickHouse using a .ini configuration file.

[capture]
dnstapsocket = tcp://0.0.0.0:9112
[clickhouse_output]
clickhouseaddress = localhost:9000
clickhouseusername =dnsmonster
clickhousepassword =########
clickhousedatabase =default
clickhouseoutputtype = 1

Configuration Details:

 I've configured DNS Monster to send data to ClickHouse at localhost:9000 with the database default, using the user dnsmonster and a configured password.

Despite these configurations, I'm not seeing any data being loaded into ClickHouse. I've checked the connectivity between DNS Monster and ClickHouse, and everything seems fine in terms of network and permissions.

Has anyone encountered a similar issue or can provide insights into troubleshooting this problem? Any help would be greatly appreciated.

Thanks in advance!

**Comment by mosajjal** [comment]
Hi. 

A few steps to consider:

1) start with stdout as the output and see if you can observe any DNS data show up. that way you can make sure your dnstap is working properly. 

2) if everything is working and you suspect the issue might be clickhouse, pay close attention to the way batching is set up for clickhouse. depending on how many packets per seconds you're intending to ingest, there are different batching strategies you can use. 

hope it helps.

**Reply by pagliardinipert** [reply]
thanks
